gflownet/envs/crystals/clattice_parameters.py

```diff
@@ -0,0 +1,494 @@
+            angles_ignored_dims = [False, True, True]
+            self.alpha_idx = 3
+            self.beta_idx = 3
+            self.gamma_idx = 3
+        # alpha != beta, alpha != gamma, beta != gamma
+        elif self.lattice_system == TRICLINIC:
+            angles_ignored_dims = [False, False, False]
+            self.alpha_idx = 3
+            self.beta_idx = 4
+            self.gamma_idx = 5
+        else:
+            raise NotImplementedError
+        self.ignored_dims = lengths_ignored_dims + angles_ignored_dims
+
+    def _step(
+        self,
+        action: Tuple[float],
+        backward: bool,
+    ) -> Tuple[List[float], Tuple[float], bool]:
+        """
+        Updates the dimensions of the state corresponding to the ignored dimensions
+        after a call to the Cube's _step().
+        """
+        state, action, valid = super()._step(action, backward)
+        for idx, (param, is_ignored) in enumerate(zip(PARAMETERS, self.ignored_dims)):
+            if not is_ignored:
+                continue
+            param_idx = self._get_index_of_param(param)
+            if param_idx is not None:
+                state[idx] = state[param_idx]
+            else:
+                state[idx] = getattr(self, f"{param}_state")
+        self.state = copy(state)
+        return self.state, action, valid
+
+    def _unpack_lengths_angles(
+        self, state: Optional[List[int]] = None
+    ) -> Tuple[Tuple, Tuple]:
+        """
+        Helper that 1) unpacks values coding lengths and angles from the state or from
+        the attributes of the instance and 2) converts them to actual edge lengths and
+        angles in the target units (angstroms or degrees).
+        """
+        state = self._get_state(state)
+
+        a, b, c, alpha, beta, gamma = [self._get_param(p) for p in PARAMETERS]
+        return (a, b, c), (alpha, beta, gamma)
+
+    def state2readable(self, state: Optional[List[int]] = None) -> str:
+        """
+        Converts the state into a human-readable string in the format "(a, b, c),
+        (alpha, beta, gamma)".
+        """
+        state = self._get_state(state)
+
+        lengths, angles = self._unpack_lengths_angles(state)
+        return f"{lengths}, {angles}"
+
+    def readable2state(self, readable: str) -> List[int]:
+        """
+        Converts a human-readable representation of a state into the standard format.
+        """
+        state = copy(self.source)
+
+        for c in ["(", ")", " "]:
+            readable = readable.replace(c, "")
+        values = readable.split(",")
+        values = [float(value) for value in values]
+
+        for param, value in zip(PARAMETERS, values):
+            state = self._set_param(state, param, value)
+        return state
+
+
+# TODO: figure out a way to inherit the (discrete) LatticeParameters env or create a
+# common class for both discrete and continous with the common methods.
+class CLatticeParametersEffectiveDim(ContinuousCube):
```

Comment:
Is there a reason to have both classes?
I get that the difference is that an instance of the CLatticePArameters class could sample LPs for any lattice system whereas, in this one, it is fixed per instance. Because of that, I get why CLatticeParameters is better suited for use in the Crystal env than the CLatticeParametersEffectiveDim. 
But in that case, I'm not sure why we also have the CLatticeParametersEffectiveDim class.

And if we do need the two different classes, given that they share a fair bit of code, I'm wondering if they should have a different inheritance structure to reduce code duplication (either inherit from some common class that would be a bit more specific than the cube, or one inherits from the other, ...)

Reply:
I don't think we need CLatticeParametersEffectiveDim. I only left it while developing the new version. Sorry for the mess.